### PR TITLE
disable batch write to tidb without locktable

### PR DESCRIPTION
**Pull request by marsishandsome**
close https://github.com/pingcap/tispark/issues/839

### CODE REVIEW

File: docs/datasource_api_userguide.md [hunk 1]
```diff
@@ -222,6 +222,23 @@ The following is TiDB-specific options, which can be passed in through `TiDBOpti
...
 TiSpark's common options can also be passed in, e.g. `spark.tispark.plan.allow_agg_pushdown`, `spark.tispark.plan.allow_index_read`, etc.
 
+## TiDB Version and Configuration for Write
+TiDB's version should >= 3.0 and make sure that the following config is correctly set.
+
+```
+# enable-table-lock is used to control table lock feature. Default is false, indicate the table lock feature is disabled.
+enable-table-lock: true
+
+# delay-clean-table-lock is used to control whether delayed-release the table lock in the abnormal situation. (Milliseconds)
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: Should we move `(Milliseconds)` to `6000`

File: docs/datasource_api_userguide.md [hunk 2]
```diff
@@ -222,6 +222,23 @@ The following is TiDB-specific options, which can be passed in through `TiDBOpti
 | writeConcurrency            | -             | false    | maximum number of threads writing data to tikv, suggest `writeConcurrency` <= 8 * `number of TiKV instance` | 0       |
 TiSpark's common options can also be passed in, e.g. `spark.tispark.plan.allow_agg_pushdown`, `spark.tispark.plan.allow_index_read`, etc.
 
+## TiDB Version and Configuration for Write
+TiDB's version should >= 3.0 and make sure that the following config is correctly set.
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: s/the following config is/the following configs are

File: docs/datasource_api_userguide.md [hunk 3]
```diff
@@ -222,6 +222,23 @@ The following is TiDB-specific options, which can be passed in through `TiDBOpti
...
+
+```
+# enable-table-lock is used to control table lock feature. Default is false, indicate the table lock feature is disabled.
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: what do you mean by saying "table lock feature"?

Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: copy from tidb https://github.com/pingcap/tidb/blob/master/config/config.toml.example#L57

File: docs/datasource_api_userguide.md (hunk 1 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: copy from tidb https://github.com/pingcap/tidb/blob/master/config/config.toml.example#L60

File: docs/datasource_api_userguide.md (hunk 3 above)
Comment (zhexuany) [reply docs/datasource_api_userguide.md]: em, I guess they did not pay attention to the comments.

Comment (birdstorm) [reply docs/datasource_api_userguide.md]:
```suggestion
# enable-table-lock is used to control table lock feature. The default value is false, indicating the table lock feature is disabled.
```

File: docs/datasource_api_userguide.md (hunk 2 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: fixed

File: docs/datasource_api_userguide.md (hunk 3 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: fixed

**Pull request merged**